common/src/include/ebbrt/MulticoreEbb.h

```diff
@@ -11,53 +12,161 @@
 
 #include <ebbrt/Cpu.h>
 #include <ebbrt/EbbAllocator.h>
+#include <ebbrt/EbbId.h>
 #include <ebbrt/EbbRef.h>
 #include <ebbrt/LocalIdMap.h>
 
 namespace ebbrt {
-template <class T, class Root> class MulticoreEbb {
+
+template <class T> class MulticoreEbbRoot {
+ public:
+  struct RepMapHandle {
+    RepMapHandle() : accessor_{std::make_unique<LocalIdMap::ConstAccessor>()} {}
+    const boost::container::flat_map<size_t, T*>* operator->() const {
+      return &operator*();
+    }
+    const boost::container::flat_map<size_t, T*>& operator*() const {
+      auto pair = boost::any_cast<std::pair<MulticoreEbbRoot<T>*, RepMap>>(
+          &(*accessor_)->second);
+      return pair->second;
+    }
+
+   private:
+    friend class MulticoreEbbRoot<T>;
+    std::unique_ptr<LocalIdMap::ConstAccessor> accessor_;
+  };
+  MulticoreEbbRoot() = default;
+  explicit MulticoreEbbRoot(EbbId id) : id_{id} {};
+
+  void SetEbbId(EbbId id) { id_ = id; }
+  RepMapHandle GetReps() const {
+    if (!id_)
+      throw std::runtime_error("No EbbId specified in MulticoreEbbRoot");
+    struct RepMapHandle rep;
+    auto found = local_id_map->Find(*rep.accessor_, id_);
+    if (!found)
```

Comment:
Should this be an assert? My thinking is that this is a logic bug, not an exceptional (but valid) outcome.